src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs

```diff
@@ -60,6 +61,10 @@ internal class QueueItem<TRequest, TResponse, TRequestContext> : IQueueItem<TReq
 
         MutatesServerState = mutatesSolutionState;
         MethodName = methodName;
+
+        var telemetryService = lspServices.GetRequiredServices<AbstractTelemetryService>().FirstOrDefault();
+
+        _requestTelemetryScope = telemetryService?.CreateRequestScope(methodName);
```

Comment:
`GetRequiredServices` says to me that the service is expected to always exist, but the `FirstOrDefault` and null check operator says to me that it isn't. Which one is it?

(and i'm hoping that the `GetRequiredServices` bit is wrong, or this will presuambly stop Razor and Web Tools from working once this is inserted)

Reply:
GetRequiredServices is poorly named (the whole ILspServices implementation is a bit wonky). It doesn't throw if there aren't any items, but rather returns an empty enumeration.

I went ahead and used this as it gave me what I need and I didn't want to change the interface, and really didn't want to change the implementation.

Reply:
Thanks, happy to hear its not actually required 😁